### DISCUSSION TITLE
Example Blink_AnalodRead.ino not working on Mega2560

### DISCUSSION THREAD

**Discussion by dlwalter** [feilipu/Arduino_FreeRTOS_Library#80] [Q&A]
**Describe the issue**
Running the example Blink_AnalogRead.ino  on an Arduino MEGA2560 with Arduino IDE 1.8.13 on macOS 10.15.6

Running the example ino results in a solid red LED with no blinking.  Running the Arduino basics example blink.ino runs fine.

**Expected behavior**
I would expect the build in LED to blink

**Desktop (please complete the following information):**
 - OS: macOS 10.15.6
 - Arduino IDE 1.8.13

**Additional context**

**Comment by dlwalter** [comment]
wrong version.  Using 10.3.0-8  Upgraded to 10.4.1 and its working correctly.

**Reply by blake5634** [reply]
Where do we get 10.4.1?   It seems 10.3.0-8 is the latest release?  I'm having same no-flash issue on ATMEGA 2560, but if I add a little flash loop into setup, Blink_AnalogRead.ino indicates that setup at least runs.

**Reply by blake5634** [reply]
OK now I see it over at freeRTOS.org, but how does one install that version with Arduino IDE?
THX

**Reply by blake5634** [reply]
I figured out the install (which involves mousing all source files from src -->  top level of the folder) and the code builds (now using 10.4.3)  but the result is the same.   When I build and upload blink_analogread.ino, I get exactly 3 flashes and then it hangs, and there is no serial data from the analogread task.    Any ideas?

**Reply by blake5634** [reply]
Working!    I don't know how you are supposed to know to ignore the comments in the official examples, but at the end of blink_analoginput.ino setup() it says
`  // Now the task scheduler, which takes over control of scheduling individual tasks, is automatically started.`
but nothing works until you add just after that comment:
`       vTaskStartScheduler();`

**Reply by feilipu** [reply]
This step should be unnecessary.

All you are doing by calling `vTaskStartScheduler()` in your initialisation rather than using the [Arduino function `initVariant()`](https://github.com/arduino/ArduinoCore-avr/blob/master/cores/arduino/main.cpp#L37) that [calls `vTaskStartScheduler()`](https://github.com/feilipu/Arduino_FreeRTOS_Library/blob/master/src/variantHooks.cpp#L57).

There is another issue underlying. It would be best to find out what that is.
Is there more you can say about the environment you are using?

**Reply by feilipu** [reply]
> I figured out the install (which involves mousing all source files from src --> top level of the folder) and the code builds (now using 10.4.3).

There is no reason to copy / paste files manually. The latest release (currently 10.4.3-8) is always available to install directly from the Arduino library manager.

![Screenshot from 2021-05-15 13-34-17](https://user-images.githubusercontent.com/3955592/118346942-b57a5000-b582-11eb-8dd7-34fa56564f30.png)

**Reply by blake5634** [reply]
I should have clarified, I'm using the IDE on Linux which does not have the library manager.

**Reply by feilipu** [reply]
The Arduino IDE is supported on 4 versions of Linux (plus MacOSX and ChromeBook), and has supported contributed libraries via the Library Manager for a very long time. I'm running the Arduino IDE on Ubuntu 20.04 currently.

Perhaps [download the Arduino IDE](https://www.arduino.cc/en/software) from their web site?
![Screenshot from 2021-05-24 10-12-17](https://user-images.githubusercontent.com/3955592/119281711-04f30700-bc7a-11eb-8403-c99456040490.png)
